### Практическое задание: Урок 1. Случайные события. Условная вероятность. Формула Байеса. Независимые испытания

##### 1) Из колоды в 52 карты извлекаются случайным образом 4 карты. a) Найти вероятность того, что все карты – крести

In [1]:
from math import factorial

In [2]:
def combinations(n, k):
    return int(factorial(n) / (factorial(k) * factorial(n - k)))

Количество крести в колоде:

In [3]:
52/4

13.0

In [4]:
m = combinations(13, 4)

In [5]:
m

715

In [6]:
n = combinations(52, 4)

In [7]:
n

270725

In [8]:
P = m/n

In [9]:
P

0.0026410564225690276

##### Ответ: вероятность того, что все карты – крести равна 0.0026410564225690276

##### б) Найти вероятность, что среди 4-х карт окажется хотя бы один туз.

In [10]:
m = combinations(48, 4)

In [11]:
m

194580

In [12]:
n = combinations(52, 4)

In [13]:
n

270725

In [14]:
P = 1 - m/n

In [15]:
P

0.2812632745405854

##### Ответ: вероятность, что среди 4-х карт окажется хотя бы один туз равна 0.2812632745405854

##### 2) На входной двери подъезда установлен кодовый замок, содержащий десять кнопок с цифрами от 0 до 9. Код содержит три цифры, которые нужно нажать одновременно.  Какова вероятность того, что человек, не знающий код, откроет дверь с первой попытки?

In [16]:
m = 1

In [17]:
n = combinations(10, 3)

In [18]:
n

120

In [19]:
P = m/n

In [20]:
P

0.008333333333333333

##### Ответ:  вероятность того, что человек, не знающий код, откроет дверь с первой попытки равна 0.008333333333333333

##### 3) В ящике имеется 15 деталей, из которых 9 окрашены. Рабочий случайным образом извлекает 3 детали. Какова вероятность того, что все извлеченные детали окрашены?

In [21]:
n = combinations(15, 3)

In [22]:
n

455

In [23]:
m = combinations(9, 3)

In [24]:
m

84

In [25]:
P = m/n

In [26]:
P

0.18461538461538463

##### Ответ:  вероятность того,что все извлеченные детали окрашены равна 0.18461538461538463

##### 4) В лотерее 100 билетов. Из них 2 выигрышных. Какова вероятность того, что 2 приобретенных билета окажутся выигрышными?

In [27]:
K = combinations(2, 2)

In [28]:
K

1

In [29]:
N = combinations(100, 2)

In [30]:
N

4950

In [31]:
L = combinations(98, 0)

In [32]:
L

1

In [33]:
P = (K*L)/N

In [34]:
P

0.00020202020202020202

##### Ответ:  вероятность того,что 2 приобретенных билета окажутся выигрышными равна 0.00020202020202020202